src/evented-tokenizer.ts

```diff
@@ -1,23 +1,32 @@
 import { preprocessInput, isAlpha, isSpace } from './utils';
 
-function EventedTokenizer(delegate, entityParser) {
-  this.delegate = delegate;
-  this.entityParser = entityParser;
-
-  this.state = null;
-  this.input = null;
-
-  this.index = -1;
-  this.line = -1;
-  this.column = -1;
-  this.tagLine = -1;
-  this.tagColumn = -1;
-
-  this.reset();
-}
+export default class EventedTokenizer {
+  private delegate: any;
+  private entityParser: any;
+  private state: any;
```

Comment:
I _think_ you can set the default value also: `private state: any = null;` (same for the other cases that are being initialized in the constructor now).